File: Scadex.Model/Dtos/Role/Queries/RoleDto.cs
using Scadex.Core.Model;

namespace Scadex.Model.Dtos.Role.Queries;

public class RoleDto : IDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    /// <summary> Sistem rolu: adi ve aktifligi degistirilemez (<c>RoleService.UpdateAsync</c> 403 doner). Izinleri yine duzenlenebilir. </summary>
    public bool IsImmutable { get; set; }
    public string? CreatedBy { get; set; }
    public string? UpdatedBy { get; set; }
    public DateTime? CreateDateUtc { get; set; }
    public DateTime? UpdateDateUtc { get; set; }
    public bool IsActive { get; set; }
}

File: Scadex.WebUI/src/models/role/queries/roleDto.ts
/** Ayna: Scadex.Model/Dtos/Role/Queries/RoleDto.cs */
export interface RoleDto {
  id: string;
  name: string;
  /** Sistem rolü: adı ve aktifliği değiştirilemez (sunucu 403 döner). İzinleri düzenlenebilir. */
  isImmutable: boolean;
  createdBy: string | null;
  updatedBy: string | null;
  /** `Z` soneki YOK (`datetime2`) — gösterilecekse `toUtcDate` ile. */
  createDateUtc: string | null;
  updateDateUtc: string | null;
  isActive: boolean;
}
